internal/logger/logger.go

```diff
@@ -1,3 +1,108 @@
 package logger
 
-// TODO: Implement logger and refactor all the log statements to use it
+import (
+	"fmt"
+	"log"
+	"log/slog"
+	"os"
+)
+
+var LogLevelString = getLogLevel()
+
+// RawLogger is the raw global logger object used for calls wrapped by the logging package
+var RawLogger = InitLogger(LogLevelString, "")
+
+// InitLogger initializes a cluster-id specific child logger
+func InitLogger(logLevelString string, clusterID string) *slog.Logger {
+	var logLevel slog.Level
+	var addSource bool
+	switch logLevelString {
+	case "debug":
+		logLevel = slog.LevelDebug
+		addSource = true
+	case "info":
+		logLevel = slog.LevelInfo
+	case "warn", "warning":
+		logLevel = slog.LevelWarn
+	case "error":
+		logLevel = slog.LevelError
+	default:
+		log.Fatalln("Invalid log level:", logLevelString)
+	}
+
+	// Create JSON handler with options
+	opts := &slog.HandlerOptions{
+		Level:     logLevel,
+		AddSource: addSource,
+	}
+	handler := slog.NewJSONHandler(os.Stdout, opts)
+	logger := slog.New(handler)
+
+	// Add structured fields using With()
+	logger = logger.With(
+		slog.String("cluster_id", clusterID),
+	)
+	return logger
+}
+
+// Info wraps slog.Logger.Info()
+func Info(args ...interface{}) {
+	RawLogger.Info(fmt.Sprint(args...))
+}
+
+// Debug wraps slog.Logger.Debug()
+func Debug(args ...interface{}) {
+	RawLogger.Debug(fmt.Sprint(args...))
+}
+
+// Warn wraps slog.Logger.Warn()
+func Warn(args ...interface{}) {
+	RawLogger.Warn(fmt.Sprint(args...))
+}
+
+// Error wraps slog.Logger.Error()
+func Error(args ...interface{}) {
+	RawLogger.Error(fmt.Sprint(args...))
+}
+
+// Fatal wraps slog.Logger.Error() and calls log.Fatal()
+func Fatal(args ...interface{}) {
+	msg := fmt.Sprint(args...)
+	RawLogger.Error(msg)
+	log.Fatal(msg)
+}
+
+// Infof wraps slog.Logger.Info() with formatted message
+func Infof(template string, args ...interface{}) {
+	RawLogger.Info(fmt.Sprintf(template, args...))
+}
+
+// Debugf wraps slog.Logger.Debug() with formatted message
+func Debugf(template string, args ...interface{}) {
+	RawLogger.Debug(fmt.Sprintf(template, args...))
+}
+
+// Warnf wraps slog.Logger.Warn() with formatted message
+func Warnf(template string, args ...interface{}) {
+	RawLogger.Warn(fmt.Sprintf(template, args...))
+}
+
+// Errorf wraps slog.Logger.Error() with formatted message
+func Errorf(template string, args ...interface{}) {
+	RawLogger.Error(fmt.Sprintf(template, args...))
+}
+
+// Fatalf wraps slog.Logger.Error() with formatted message and calls log.Fatal()
+func Fatalf(template string, args ...interface{}) {
+	msg := fmt.Sprintf(template, args...)
+	RawLogger.Error(msg)
+	log.Fatal(msg)
+}
+
+// getLogLevel returns the log level from the environment variable LOG_LEVEL
+func getLogLevel() string {
+	if envLogLevel, exists := os.LookupEnv("LOG_LEVEL"); exists {
```

Comment:
We should use Viper to configure the Log Level. Then we'll get automatic Env var lookups. 

It's already an option:
https://github.com/rhobs/rhobs-synthetics-agent/blob/main/cmd/agent/main.go#L53

```
func getLogLevel() string {
	// viper.GetString will check the flag, environment variable, or config file
	// in that order, based on the BindPFlag call and other viper settings.
	if level := viper.GetString("log_level"); level != "" {
		return level
	}
	// Fallback to a default value if no level is found.
	return "info"
}
```